### PR TITLE
KAFKA-16202: Extra dot in error message in producer

### PR DESCRIPTION
**Pull request by infantlikesprogramming**
The author of KAFKA-16202 noticed that there is an extra dot in the error message for KafkaStorageException message.

Looking into org.apache.kafka.clients.producer.internals.Sender, it turns out that the string for the message to be sent in completeBatch() added an extra dot. I think that the formatted component (error.exception(response.errorMessage).toString())) of the error message already has a dot in the end of its string. Thus the dot after the "{}" sign caused the extra dot.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by infantlikesprogramming** [comment]
@mimaison 
Would you mind reviewing my PR?
Thank you

File: clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java [hunk 1]
```diff
@@ -704,7 +704,7 @@ private void completeBatch(ProducerBatch batch, ProduceResponse.PartitionRespons
                             "topic-partition may not exist or the user may not have Describe access to it",
                         batch.topicPartition);
                 } else {
-                    log.warn("Received invalid metadata error in produce request on partition {} due to {}. Going " +
+                    log.warn("Received invalid metadata error in produce request on partition {} due to {} Going " +
                             "to request metadata update now", batch.topicPartition,
```
Comment (appchemist) [inline clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java]: Some error have dot, while others don't.
So, When dot is absent, there is no dot before 'Going'.

See `Errors.INCONSISTENT_TOPIC_ID` for details.

Comment (infantlikesprogramming) [reply clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java]: Thank you so much for your comment. I have made changes based on your review in the latest (4th) commit.

**Review by infantlikesprogramming — COMMENTED** [review]
@appchemist 
Would you mind reviewing my new commit?
I have created a new variable for the error message and make appropriate change to the variable in case it does not end with ".".

File: clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java (hunk 1 above)
Comment (appchemist) [reply clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java]: How about including dot in Errors's `defaultExceptionString` that don't currently include dot?
I referred to [Use grammatically correct error messages](https://learn.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions?redirectedfrom=MSDN#use-grammatically-correct-error-messages).
In addition, most Errors's `defaultExceptionString` already include dot.

Comment (infantlikesprogramming) [reply clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java]: This makes a lot of sense. Thank you so much @appchemist. It took me a while to understand the code. I will fix this in my next commit.

**Review by infantlikesprogramming — COMMENTED** [review]
@appchemist I have added full stops to defaultExceptionString for errors that lacked it. Also, I have made appropriate change to the log message in Sender

**Review by appchemist — APPROVED** [review]

**Review by appchemist — APPROVED** [review]

**Comment by chia7712** [comment]
@infantlikesprogramming do you have jira account? I try to assign the jira to you

**Comment by infantlikesprogramming** [comment]
> @infantlikesprogramming do you have jira account? I try to assign the jira to you
@chia7712 thank you so much for asking. I have tried to sign up for a JIRA account but got no reply. I have just submitted another sign-up form on JIRA

**Comment by chia7712** [comment]
@infantlikesprogramming I just approve your request. Will assign the jira to you

**Comment by infantlikesprogramming** [comment]
@chia7712 Thank you so much. I have just created a JIRA account. Also, would you mind giving me some suggestions on what should do to get my PR merged? Thank you.

**Comment by chia7712** [comment]
@infantlikesprogramming this PR is in my queue. If you want to do more contribution, please feel free to look up other tickets before reviewing.

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by infantlikesprogramming** [comment]
> @infantlikesprogramming this PR is in my queue. If you want to do more contribution, please feel free to look up other tickets before reviewing.

@chia7712 Understood. Thank you!

**Pull request merged**